test/extended/csrapprover/csrapprover.go

```diff
@@ -0,0 +1,163 @@
+package csrapprover
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	certv1beta1 "k8s.io/api/certificates/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/client-go/kubernetes/scheme"
+	certclientv1beta1 "k8s.io/client-go/kubernetes/typed/certificates/v1beta1"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+func init() {
+	certv1beta1.AddToScheme(scheme.Scheme)
+}
+
+var _ = g.Describe("don't pwn cluster via MCS and autoapprover:", func() {
+	// since login mutates the current kubeconfig we want to use NewCLI
+	// as that will give each one of our test runs a new config via SetupProject
+	oc := exutil.NewCLI("cluster-client-cert", exutil.KubeConfigPath())
+	defer g.GinkgoRecover()
+
+	g.It("deny pod's access to /config/master API endpoint", func() {
```

Comment:
if you create a pod with host network, this should be able to read the metadata endpoint right?

Reply:
Per https://github.com/openshift/origin/pull/22821#issue-278078899, I don't think it should

Reply:
@stlaz is correct, it is blocked from host network too to the MCS ports.

Reply:
> @stlaz is correct, it is blocked from host network too to the MCS ports.

Wait so how does _anyone_ access that port?

Reply:
> In conjunction with the MCS/approver changes happening elsewhere, this attempts to block all (non-privileged / non-`CAP_NET_ADMIN`) pods from being able to reach the MCS.

Oh, I think I get it.

Reply:
Ignition runs from the system's initramfs, long before any pods (such as the SDN) have landed on the machine.